NMS-14616: Quick Start - Performance Data Gathering (#5263)
OpenNMS/opennms: docs/modules/operation/pages/deep-dive/performance-data-collection/resource-types.adoc

----

== Persistence selector strategies

.Persistence selector strategies
[options="header" "autowidth", cols="1,2"]
|===
| Class
| Description

| `org.opennms.netmgt.collection.support.PersistAllSelectorStrategy`
| Persist all indexes.

| `org.opennms.netmgt.collection.support.PersistRegexSelectorStrategy`
| Persist indexes based on JEXL evaluation.
|===

=== PersistRegexSelectorStrategy

The PersistRegexSelectorStrategy class takes a single parameter, `match-expression`, which defines a JEXL expression.
On evaluation, this expression should return either true (persist index to storage) or false (discard data).

== Storage strategies

.Storage strategies
[options="header" "autowidth", cols="1,2"]
|===
| Class
| Storage Path Value

| `org.opennms.netmgt.collection.support.IndexStorageStrategy`
| Index

| `org.opennms.netmgt.collection.support.JexlIndexStorageStrategy`
| Value after JexlExpression evaluation.

| `org.opennms.netmgt.collection.support.ObjectNameStorageStrategy`
| Value after JexlExpression evaluation.

| `org.opennms.netmgt.dao.support.FrameRelayStorageStrategy`

> | interface label + '.' + dlci

Reviewer: Let's capitalize DLCI

Author: ```suggestion
| interface label + '.' + DLCI
```

| `org.opennms.netmgt.dao.support.HostFileSystemStorageStrategy`

> | Uses the value from the hrStorageDescr column in the hrStorageTable, cleaned up for UNIX file systems.

Reviewer: Let's make it "cleaned up to make a valid directory name on supported file systems". These paths should work on more than just UNIX file systems.

Author: ```suggestion
| Uses the value from the hrStorageDescr column in the hrStorageTable, cleaned to make a valid directory name on supported file systems.
```

| `org.opennms.netmgt.dao.support.SiblingColumnStorageStrategy`

> | Uses the value from an SNMP lookup of OID in sibling-column-name parameter, cleaned up for UNIX file systems.

Reviewer: Let's make it "cleaned up to make a valid directory name on supported file systems". These paths should work on more than just UNIX file systems. (repeat comment)

Author: ```suggestion
| Uses the value from an SNMP lookup of OID in sibling-column-name parameter, cleaned to make a valid directory name on supported file systems.
```

| `org.opennms.protocols.xml.collector.XmlStorageStrategy`

> | Index, but cleaned up for UNIX file systems.

Reviewer: Let's make it "cleaned up to make a valid directory name on supported file systems". These paths should work on more than just UNIX file systems. (repeat repeat comment)

Author: ```suggestion
| Index, cleaned to make a valid directory name on supported file systems.
```

=== SiblingColumnStorageStrategy

[options="header" "autowidth", cols="1,2"]
|===
| Parameter
| Description

| sibling-column-name

> | Alternate string value to use for index.

Reviewer: Name of another `mibObj` in the same `group`, whose type must be `string`. This object's value will form the starting point for the index.

Author: ```suggestion
| The name of another `mibObj` in the same group.
This string value forms the starting point for the index.
```

| replace-first

> | Regex pattern, replaces only the first match.

Reviewer: Specifies a regex replacement to perform against the value of the object pointed to by `sibling-column-name`. Only the first match in the input will be replaced.

Author: ```suggestion
| Specifies a regex replacement to perform against the value of the object that `sibling-column-name` points to.
Only the first match in the input is replaced.
```

| replace-all

> | Regex pattern, replaces all matches.

Reviewer: Specifies a global regex replacement to perform against the value of the object pointed to by `sibling-column-name`. Every match in the input will be replaced.

Author: ```suggestion
| Specifies a global regex replacement to perform against the value of the object that `sibling-column-name` points to.
Every match in the input is replaced.
```
